sympy/combinatorics/fp_groups.py

```diff
@@ -1003,4 +1005,97 @@ def first_in_class(C, Y=[]):
     return True
 
 
+# Pg 175 [1]
+def define_schreier_generators(C):
+    y = []
+    gamma = 1
+    f = C.fp_group
+    X = f.generators
+    C.P = [[None]*len(C.A) for i in range(C.n)]
+    for alpha, x in product(C.omega, C.A):
+        beta = C.table[alpha][C.A_dict[x]]
+        if beta == gamma:
+            C.P[alpha][C.A_dict[x]] = "<identity>"
+            C.P[beta][C.A_dict_inv[x]] = "<identity>"
+            gamma += 1
+        elif x in X and C.P[alpha][C.A_dict[x]] is None:
+            y_alpha_x = '%s_%s' % (x, alpha)
+            y.append(y_alpha_x)
+            C.P[alpha][C.A_dict[x]] = y_alpha_x
+    grp_gens = list(free_group(', '.join(y)))
+    C.schreier_free_group = grp_gens.pop(0)
+    C.schreier_generators = grp_gens
+    # replace all elements of P by, free group elements
+    for i, j in product(range(len(C.P)), range(len(C.A))):
+        # if equals "<identity>", replace by identity element
+        if C.P[i][j] == "<identity>":
+            C.P[i][j] = C.schreier_free_group.identity
+        elif isinstance(C.P[i][j], str):
+            r = C.schreier_generators[y.index(C.P[i][j])]
+            C.P[i][j] = r
+            beta = C.table[i][j]
+            C.P[beta][j + 1] = r**-1
+
+
+def reidemeister_relators(C):
+    R = C.fp_group.relators()
+    rels = set([rewrite(C, coset, word) for word in R for coset in range(C.n)])
+    identity = C.schreier_free_group.identity
+    order_1_gens = set([i for i in rels if len(i) == 1])
```

Comment:
I am not sure, whether simplifications like these should be kept just here of move under the `Tietze transformation program` since I have studied the "presentation by generators", if "presentation by generators" also expects such simplification then we move this out.